### ISSUE TITLE
Improve and unify the RESP error message

### ISSUE THREAD

**Issue by git-hulk** [apache/kvrocks#2335]
### Search before asking

- [X] I had searched in the [issues](https://github.com/apache/kvrocks/issues) and found no similar issues.

### Motivation

As mentioned in the comment: https://github.com/apache/kvrocks/blob/unstable/tests/gocase/util/server.go#L142, we should take care of different RESP error kinds to avoid introducing the protocol compatibility issues. 

### Solution

Use an unify function to return the RESP error message, like the following:

```c++
const std::string redis::Error(ErrorKind kind, const std::string &message) {
  switch(kind) {
    case ErrorKind::Loading: 
    case ErrorKind::NoProto:
    ...
    default:
  }
}
```

### Are you willing to submit a PR?

- [X] I'm willing to submit a PR!

**Comment by PragmaTwice** [comment]
The problem is that we add ERR prefix for all msg returned by `Status`.

One solution is to add a new status code for no ERR prefix. (Or, add many status code for LOADING, NOPROTO...)

**Comment by git-hulk** [comment]
> The problem is that we add ERR prefix for all msg returned by Status.

Yes

> One solution is to add a new status code for no ERR prefix. (Or, add many status code for LOADING, NOPROTO...)

This also sounds good

**Comment by git-hulk** [comment]
closes by #2362
